donjo-app/models/Keuangan_grafik_manual_model.php

```diff
@@ -0,0 +1,812 @@
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_pendapatan2($d['Jenis'], $thn);
+		  $data[$i]['realisasi'] = $this->real_pendapatan2($d['Jenis'], $thn);
+		}
+		return $data;
+  }
+
+  private function sub_belanja2($kelompok, $thn=false)
+  {
+		$this->db->select('Kelompok, Jenis, Nama_Jenis');
+		$this->db->where('Kelompok', $kelompok);
+
+		$data = $this->db->get('keuangan_manual_ref_rek3')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_belanja2($d['Jenis'], $thn);
+		  $data[$i]['realisasi'] = $this->real_belanja2($d['Jenis'], $thn);
+		}
+		return $data;
+  }
+
+  private function sub_pembiayaan2($kelompok, $thn=false)
+  {
+		$this->db->select('Kelompok, Jenis, Nama_Jenis');
+		$this->db->where('Kelompok', '6.1.');
+		$this->db->where('Kelompok', $kelompok);
+
+		$data = $this->db->get('keuangan_manual_ref_rek3')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_pembiayaan2($d['Jenis'], $thn);
+		  $data[$i]['realisasi'] = $this->real_pembiayaan2($d['Jenis'], $thn);
+		}
+		return $data;
+  }
+
+  private function sub_pembiayaan_keluar2($kelompok, $thn=false)
+  {
+		$this->db->select('Kelompok, Jenis, Nama_Jenis');
+		$this->db->where('Kelompok', '6.2.');
+		$this->db->where('Kelompok', $kelompok);
+
+		$data = $this->db->get('keuangan_manual_ref_rek3')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_pembiayaan_keluar2($d['Jenis'], $thn);
+		  $data[$i]['realisasi'] = $this->real_pembiayaan_keluar2($d['Jenis'], $thn);
+		}
+		return $data;
+  }
+
+  private function pagu_pendapatan2($jenis, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 6) AS Jenis, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $jenis, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Jenis');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function pagu_belanja2($jenis, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 6) AS Jenis, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $jenis, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Jenis');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function pagu_pembiayaan2($jenis, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 6) AS Jenis, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $jenis, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Jenis');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function pagu_pembiayaan_keluar2($jenis, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 6) AS Jenis, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $jenis, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Jenis');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
```

Comment:
Apa beda kedua ini?

Reply:
Kedua fungsi itu berbeda karena pada fungsi akhir akan berbeda jenis kelompok yang dipakai

**Pembiayaan Masuk**
pagu_pembiayaan2 -> sub_pembiayaan2 : $this->db->where('Kelompok', '6.1.');

**Pembiayaan Keluar**
pagu_pembiayaan_keluar2 -> sub_pembiayaan_keluar2 : $this->db->where('Kelompok', '6.2.');

Reply:
Lihat komentar untuk kasus di bawah.